pkg/auth/authenticator.go

```diff
@@ -69,3 +71,46 @@ func (ba *BuiltinAuthenticator) AuthenticateUser(ctx context.Context, username,
 func (ba *BuiltinAuthenticator) String() string {
 	return "built in authenticator"
 }
+
+// GetOrCreateUser searches for the user by username, and if not found, creates a new user with the given username and
+// external user identifier.  It returns the user.
+// This function is meant to be used by Authenticator implementations in the ChainAuthenticator.
+func GetOrCreateUser(ctx context.Context, authService Service, friendlyName, identifier, defaultUserGroup, source string) (*model.User, error) {
```

Comment:
- identifier is not clear, we use username mostly (sometimes userId meh) 
- ordering, makes more sense that username is first 

```suggestion
func GetOrCreateUser(ctx context.Context, authService Service, username, friendlyName, defaultUserGroup, source string) (*model.User, error) {
```